ARCs/arc-0056.md

```diff
@@ -161,20 +159,32 @@ interface Method {
   desc?: string;
   /** The arguments of the method, in order */
   args: Array<{
-    /** The type of the argument */
+    /** The type of the argument. The `struct` field should also be checked to determine if this arg is a struct. */
     type: ABIType;
     /** If the type is a struct, the name of the struct */
     struct?: StructName;
     /** Optional, user-friendly name for the argument */
     name?: string;
     /** Optional, user-friendly description for the argument */
     desc?: string;
-    /** The default value that clients should use. MUST be base64 encoded bytes */
-    defaultValue?: string;
+    /** The default value that clients should use. */
+    defaultValue?: {
+      /** Base64 encoded bytes or uint64 */
+      data: string | bigint;
+      /** How the data is encoded. This is the encoding for the data provided here, not the arg type */
+      type: ABIType | AVMType;
+      /** Where the default value is coming from
+       * - box: The data key signifies the box key to read the value from
+       * - global: The data key signifies the global state key to read the value from
+       * - local: The data key signifies the local state key to read the value from (for the sender)
+       * - literal: the value is a literal and should be passed directly as the argument
```

Comment:
As a suggestion, it could be the method name or method signature and the client is expected to use default values to populate any arguments. That implementation is simple, but powerful and I have an implementation ready to go (which is what it did for ARC-32).

Reply:
Added in 3476dad